nix-support/third-party-python-packages.nix

```diff
@@ -0,0 +1,44 @@
+self: super: {
+  pybind11_json = super.stdenv.mkDerivation {
+    name = "pybind11_json";
+    src = super.fetchFromGitHub {
+      owner = "pybind";
+      repo = "pybind11_json";
+      rev = "0.2.13";
+      sha256 = "sha256:Kl/QflV2bBoH72/LW03K8JDlhBF+DYYXL47A5s1nmTw=";
+    };
+    nativeBuildInputs = [ super.cmake ];
+    buildInputs = [ super.python3Packages.pybind11 super.nlohmann_json ];
+  };
+  qwasm = super.python3.pkgs.buildPythonPackage {
+    name = "qwasm";
+    src = super.fetchFromGitHub {
+      owner = "CQCL";
+      repo = "qwasm";
+      rev = "35ebe1e2551449d97b9948a600f8d2e4d7474df6";
```

Comment:
There's likely to be a certain amount of bitrot here as dependency versions get updated; maybe we could prevent this with some ugly CI checks but probably not worth it for now.